src/Chainweb/Pact/PactService.hs

```diff
@@ -439,14 +441,19 @@ validateChainwebTxsPreBlock dbEnv cp bh hash txs = do
       m <- readCoinAccount dbEnv sender
       case m of
         Nothing -> return True
-        Just (T2 b _g) -> return $! b >= limitInCoin
+        Just (T2 b _g) -> do
+          let validations = and $ map ($ tx) validators
+          return $! b >= limitInCoin && validations
 
     checkOne tx = do
         let pactHash = view P.cmdHash tx
         mb <- _cpLookupProcessedTx cp pactHash
         if mb == Nothing
-        then checkAccount tx
+        -- prop_tx_ttl_newblock_validate
+        then checkAccount [k] tx
         else return False
+      where
+        k tx' = timingsCheck blockOriginationTime (payloadObj <$> tx')
```

Comment:
Need to guard this w/ `blockHeight /= 0` also.

Reply:
done